File: benchmarks/oopsla25/subjects/067.py
import sys

def f067(s: str):
  assert len(s) <= 3
  return s[0:3]

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f067(my_string)

File: benchmarks/oopsla25/subjects/067.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

const char* f067(const char *s) {
    assert(strlen(s) <= 3);
    
    static char result[4];
    strncpy(result, s, 3);
    result[3] = '\0';
    
    return result;
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f067(my_string);
  return 0;
}
